### ISSUE TITLE
cosine reward in GRPO is inconsistent with expectations

### ISSUE THREAD

**Issue by youyc22** [modelscope/ms-swift#3353]
**Describe the bug**

When training GRPO with cosine reward, it was found that the length of the correct answer and the reward were inconsistent with expectations (the shorter the answer, the higher the reward). The reason was that the definitions of cosine_max_len and max_completion_length were inconsistent. The former was the number of characters, i.e. len(content), while the latter was the number of tokens. The code defaulted to cosine_max_len=max_completion_length

在使用cosine奖励的GRPO训练时，发现正确答案的奖励和长度关系毫无规律，与预期不一致（越短奖励越高），发现原因是cosine_max_len和max_completion_length的定义不一致，前者是字符的数量，即len(content)，而后者是token数量，而代码默认cosine_max_len=max_completion_length，希望使用分词器计算长度，或者在文档、readme中给出说明，防止更多的人遇到这个问题

#3358 

![Image](https://github.com/user-attachments/assets/02bf003e-5074-45f8-a344-509502b61da6)

![Image](https://github.com/user-attachments/assets/53a95709-cb7f-44b2-8453-a26d2b01b13b)

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)

training configs：
训练配置：
```
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NPROC_PER_NODE=4  \
swift rlhf \
    --rlhf_type grpo \
    --model /path/to/DeepSeek-R1-Distill-Qwen-14B \
    --reward_funcs cosine  \
    --cosine_max_len 12000 \
    --cosine_max_len_value_correct 0.4 \
    --cosine_min_len_value_correct 0.8 \
    --cosine_max_len_value_wrong 0 \
    --cosine_min_len_value_wrong -0.2 \
    --train_type lora \
    --use_lmdeploy true \
    --lora_rank 32 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --torch_dtype bfloat16 \
    --dataset 'final-final.jsonl' \
    --max_completion_length 12000 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-4 \
    --gradient_accumulation_steps 4 \
    --eval_steps 80 \
    --save_steps 80 \
    --save_total_limit 10 \
    --logging_steps 1 \
    --report_to wandb \
    --max_length 16000 \
    --output_dir trained_14b \
    --warmup_ratio 0.1 \
    --num_generations 4 \
    --temperature 0.9 \
    --top_p 0.95 \
    --top_k 50 \
    --repetition_penalty 1.1 \
    --async_generate true \
    --deepspeed zero2 \
    --system 'prompt.txt' \
    --log_completions true \
    --num_iterations 1 \
    --num_infer_workers 4 \
```

**Additional context**
Add any other context about the problem here(在这里补充其他信息)
